test/klippy/pwm.test

```diff
@@ -28,3 +28,11 @@ SET_PIN PIN=soft_pwm_pin VALUE=0.5 CYCLE_TIME=0.5
 SET_PIN PIN=soft_pwm_pin VALUE=0.5 CYCLE_TIME=0.5
 SET_PIN PIN=soft_pwm_pin VALUE=0.75 CYCLE_TIME=0.5
 SET_PIN PIN=soft_pwm_pin VALUE=0.75 CYCLE_TIME=0.75
+
+# PWM tool
+# Basic test
+SET_PIN PIN=test_pwm_tool VALUE=0
+SET_PIN PIN=test_pwm_tool VALUE=0.5
+SET_PIN PIN=test_pwm_tool VALUE=0.5
+SET_PIN PIN=test_pwm_tool VALUE=0.25
+SET_PIN PIN=test_pwm_tool VALUE=1
```

Comment:
Other things worthy to test: 
 - `maximum_mcu_duration` by activating the pin to some non-default value and starting a toolhead move that will take longer than this duration.
 - small rasterized image with some speed and reasonable resolution to test a "real" world example (perhaps the *Klipper3D* logo)

If wanted, I can provide these tests as I used something similar (but manually).